integration-app/src/main/java/uk/gov/crowncommercial/esourcing/integration/app/SalesforceConfiguration.java

```diff
@@ -1,24 +1,171 @@
+import org.springframework.core.io.buffer.DefaultDataBufferFactory;
+import org.springframework.http.MediaType;
+import org.springframework.http.ReactiveHttpInputMessage;
+import org.springframework.http.codec.HttpMessageEncoder;
+import org.springframework.http.codec.json.Jackson2JsonDecoder;
+import org.springframework.http.codec.json.Jackson2JsonEncoder;
+import org.springframework.security.oauth2.client.AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager;
+import org.springframework.security.oauth2.client.InMemoryReactiveOAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.OAuth2AuthorizationContext;
+import org.springframework.security.oauth2.client.ReactiveOAuth2AuthorizedClientProvider;
+import org.springframework.security.oauth2.client.ReactiveOAuth2AuthorizedClientProviderBuilder;
+import org.springframework.security.oauth2.client.ReactiveOAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.endpoint.WebClientReactivePasswordTokenResponseClient;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.InMemoryReactiveClientRegistrationRepository;
+import org.springframework.security.oauth2.client.registration.ReactiveClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServerOAuth2AuthorizedClientExchangeFilterFunction;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.util.MimeType;
+import org.springframework.util.MimeTypeUtils;
+import org.springframework.web.reactive.function.BodyExtractor;
+import org.springframework.web.reactive.function.BodyExtractors;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+import org.springframework.web.reactive.function.client.ExchangeStrategies;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
 import uk.gov.crowncommercial.esourcing.salesforce.client.ApiClient;
+import uk.gov.crowncommercial.esourcing.salesforce.client.RFC3339DateFormat;
 import uk.gov.crowncommercial.esourcing.salesforce.client.RfxStatusListApi;
 
 @Configuration
 public class SalesforceConfiguration {
   private static final Logger LOGGER = LoggerFactory.getLogger(SalesforceConfiguration.class);
 
-  @Value("${ccs.esourcing.salesforce-client-url}")
+  @Value("${ccs.esourcing.salesforce.client-url}")
   private String salesforceClientUrl;
 
+  @Value("${salesforce.oauth2.username}")
+  private String username;
+
+  @Value("${salesforce.oauth2.password}")
+  private String password;
+
+  @Value("${salesforce.oauth2.token.expires-in}")
```

Comment:
There is scope to add a default value so this value does not have to be specified in the application.properties. Not sure if having a default value is a good idea though?